### ISSUE TITLE
[BUG] mlflow gc - No such file or directory

### ISSUE THREAD

**Issue by sephib** [mlflow/mlflow#3595]
### Willingness to contribute
- [x] Yes. I would be willing to contribute a fix for this bug with guidance from the MLflow community.

### System information
- **Have I written custom code (as opposed to using a stock example script provided in MLflow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Suse
- **MLflow installed from (source or binary)**: binary
- **MLflow version (run ``mlflow --version``)**: 1.10
- **Python version**:
- **Exact command to reproduce**:
mlflow gc --backend-store-uri "sqlite:////path/to/sqlit3.db"

### Describe the problem
When run [gc](https://www.mlflow.org/docs/latest/cli.html#mlflow-gc) - if some of the runs were manually deleted from the artifact folder/location then the entire gc command fails (or until the 'ghosted' run is encountered).

### Other info / logs
A possible workaround/solution is to add a try and except to [local_artifact_repo](https://github.com/mlflow/mlflow/blob/master/mlflow/store/artifact/local_artifact_repo.py#L110)   
```python
try:
    shutil.rmtree(local_file_uri_to_path(artifact_path))
except OSError as e:
    print(e)
```

### What component(s), interfaces, languages, and integrations does this bug affect?
Components 
- [x] `area/artifacts`: Artifact stores and artifact logging

Interface 
- [X] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry

**Comment by dmatrix** [comment]
@sephib Thanks for filing this. When you say some runs were manually deleted, you mean using `/bin/rm -f -r <run_id>`,  instead of 
`mlflow.delete_run(run_id)`, running `mlflow gc .... ` fails or throws an exception.

Your suggestion is to print an error, informing that relevant for that particular ghosted run were not found, and continue?

cc: @harupy

**Comment by sephib** [comment]
> When you say some runs were manually deleted, you mean using /bin/rm -f -r <run_id>, instead of
> mlflow.delete_run(run_id), running mlflow gc .... fails or throws an exception.    

Yes - `mlflow gc` fails when the code encounters a non-existing folder/file that it is trying to remove.   

My suggestion is to allow the code to gracefully continue to execute the deletion of the runs while printing out the  runs that it failed to delete (and throws an exception).

**Comment by dmatrix** [comment]
> My suggestion is to allow the code to gracefully continue to execute the deletion of the runs while printing out the runs that it failed to delete (and throws an exception).

Do we want to throw an exception? I suggest we catch the exception while trying to remove the ghost run_ids, print an informative message, and carry on. Alternative, provide an option to gc [--failfast] which just fails and resorts to the original behavior.

**Comment by sephib** [comment]
> Do we want to throw an exception? I suggest we catch the exception while trying to remove the ghost run_ids, print an informative message, and carry on.

Agree  -  
How about this message :   
```
try:
    shutil.rmtree(local_file_uri_to_path(artifact_path))
except OSError as e:
    print(f'Failed to permanently delete artifacts from {artifact_path}\n {e}')
```

**Comment by dmatrix** [comment]
Perhaps, something like this... for catching and printing the message.

```
import shutil

artifact_path = "artifacts/models/delete_file.mdl"
if __name__ == "__main__":

    try:
        shutil.rmtree(artifact_path)
    except OSError as err:
        print('{}. Failed to permanently delete artifacts. Continuing with gc ...'.format(err))

[Errno 2] No such file or directory: 'artifacts/models/delete_file.mdl'. Failed to permanently delete artifacts. Continuing with gc ...

**Comment by floriancircly** [comment]
Any updates when to expect the fix? I'm using mlflow, version 1.27.0 and there is still no try/except handling in /mlflow/store/artifact/local_artifact_repo.py when deleting artefacts (mlflow gc).

**Comment by ReHoss** [comment]
please

**Comment by simonlsk** [comment]
BTW 
```
shutil.rmtree(artifact_path)
```
Fails on single file deletes using the REST API to delete artifacts. 
It should be:
```
if os.path.isfile(artifact_path):
    os.remove(artifact_path)
else:
    shutil.rmtree(artifact_path)
```
